pkg/client/client_test.go

```diff
@@ -0,0 +1,190 @@
+					},
+				},
+			}),
+		},
+		{
+			name:           "subjects change",
+			expectedUpdate: true,
+			clientset: fake.NewSimpleClientset(&rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "prometheus-k8s",
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     "prometheus-k8s",
+				},
+				Subjects: []rbacv1.Subject{
+					rbacv1.Subject{
+						Kind:      "ServiceAccount",
+						Name:      "prometheus-k8s",
+						Namespace: "openshift-monitoring",
+					},
+					rbacv1.Subject{
+						Kind:      "ServiceAccount",
+						Name:      "prometheus-k8s-new",
+						Namespace: "openshift-monitoring",
+					},
+				},
+			}),
+		},
+		{
+			name:           "labels change",
+			expectedUpdate: true,
+			clientset: fake.NewSimpleClientset(&rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "prometheus-k8s",
+					Labels: map[string]string{
+						"label": "value",
+					},
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     "prometheus-k8s",
+				},
+				Subjects: []rbacv1.Subject{
+					rbacv1.Subject{
+						Kind:      "ServiceAccount",
+						Name:      "prometheus-k8s",
+						Namespace: "openshift-monitoring",
+					},
+					rbacv1.Subject{
+						Kind:      "ServiceAccount",
+						Name:      "prometheus-k8s-new",
+						Namespace: "openshift-monitoring",
+					},
+				},
+			}),
+		},
+		{
+			name:           "annotations change",
+			expectedUpdate: false,
+			clientset: fake.NewSimpleClientset(&rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "prometheus-k8s",
+					Annotations: map[string]string{
+						"annotation": "value",
+					},
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     "prometheus-k8s",
+				},
+				Subjects: []rbacv1.Subject{
+					rbacv1.Subject{
+						Kind:      "ServiceAccount",
+						Name:      "prometheus-k8s",
+						Namespace: "openshift-monitoring",
+					},
+				},
+			}),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(st *testing.T) {
+			f := manifests.NewFactory("openshift-monitoring", "openshift-user-workload-monitoring", manifests.NewDefaultConfig())
+			crb, err := f.PrometheusK8sClusterRoleBinding()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			var c Client
+			c.kclient = tc.clientset
+			before, err := c.kclient.RbacV1().ClusterRoleBindings().Get(context.TODO(), "prometheus-k8s", metav1.GetOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			err = c.CreateOrUpdateClusterRoleBinding(crb)
+			if err != nil {
+				t.Fatal(err)
+			}
+			after, err := c.kclient.RbacV1().ClusterRoleBindings().Get(context.TODO(), "prometheus-k8s", metav1.GetOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			updated := reflect.DeepEqual(before, after)
```

Comment:
nit: the value of `updated` will be true if `before` and `after` do not differ.
however, it will be `false` if `before` and `after` do differ.
shouldn't we call it rather `unchanged`?

Reply:
makes sense, fixed